File: problems/1_palindrome/tylert/solution.js
// Run: `node solution.js`
// Redirect output to file: `node solution.js > output.file`
// Check if correct: `diff output.file ../output.txt`

const fs = require('fs');
const inputFile = fs.readFileSync("../input.txt", "utf8");
const lines = inputFile.split("\n");

function isPalindrome(string) {
  for(let i = 0; i < string.length / 2; i++) {
    const beginChar = string.charAt(i);
    const endChar = string.charAt((string.length - 1) - i);
    if (beginChar !== endChar) {
      return false;
    }
  }
  return true;
}

for (const line of lines) {
  const result = isPalindrome(line) ? "true" : "false";
  console.log(result)
}

File: problems/1_palindrome/tylert/solution.java
// Run: `javac solution.java && java solution`
// Redirect output to file: `javac solution.java && java solution > output.file`
// Check if correct: `diff output.file ../output.txt`

import java.io.FileNotFoundException;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.ArrayList;
import java.util.stream.Collectors;


class solution {
  public static List<String> readFile(String filename) {
    List<String> lines = new ArrayList<String>();
    try {
      lines = Files.lines(Paths.get(filename)).collect(Collectors.toList());
    } catch (FileNotFoundException e) {
      System.out.println("File not found");
    } catch (IOException e) {
      System.out.println("IO Error");
    }
    return lines;
  }

  public static boolean is_palindrome(String str) {
    String reversedString = "";
    for (int i = str.length() - 1; i >= 0; i--) {
      char c = str.charAt(i);
      reversedString += c;
    }
    return str.equals(reversedString);
  }

  public static void main(String[] args) {
    List<String> lines = readFile("../input.txt");
    for (String line : lines) {
      Boolean result = is_palindrome(line);
      if (result) {
        System.out.println("true");
      } else {
        System.out.println("false");
      }
    }
  }
}